### PR TITLE
Add basic test cases for gui interface

### PR DESCRIPTION
**Pull request by forslund**
## Description
Add test cases for the gui interface.

## How to test
Ensure the unittests pass

## Contributor license agreement signed?
CLA [ Yes ]

### CODE REVIEW

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2697))

File: test/unittests/enclosure/test_gui.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        self.gui.show_html(html)
+
+        sent_message = self.mock_skill.bus.emit.call_args_list[-1][0][0]
+        page_path = resolve_resource_file('ui/SYSTEM_HtmlFrame.qml')
+        page_url = 'file://{}'.format(page_path)
+        self.assertEqual(sent_message.data['page'], [page_url])
+        self.assertEqual(self.gui['html'], html)
+
+    def test_send_event(self):
+        """Check that send even sends in the correct format."""
```
Comment (krisgesling) [inline test/unittests/enclosure/test_gui.py:110]: tiny typo: even > event

File: test/unittests/enclosure/test_gui.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        self.mock_skill.bus.wait_for_response.return_value = response
+        self.assertFalse(self.gui.connected)
+
+    def test_connected(self):
+        response = Message('dummy', data={'connected': True})
+        self.mock_skill.bus.wait_for_response.return_value = response
+        self.assertTrue(self.gui.connected)
+
+    def test_connected_no_response(self):
+        """Assure that a timeout response results in not connected."""
```
Comment (krisgesling) [inline test/unittests/enclosure/test_gui.py:149]: Assure > Ensure

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2697))

**Review by krisgesling — CHANGES_REQUESTED** [review]
These are awesome, thanks so much!

A couple of minor things, but they look great overall.

File: test/unittests/enclosure/test_gui.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        expected_page = 'file:///test/ui/vogon_poetry.qml'
+        self.assertEqual(sent_message.data['page'], [expected_page])
+
+    def test_show_image(self):
+        self.gui.show_image('arthur_dent.jpg')
+
+        sent_message = self.mock_skill.bus.emit.call_args_list[-1][0][0]
+        image_path = resolve_resource_file('ui/SYSTEM_ImageFrame.qml')
+        image_url = 'file://{}'.format(image_path)
+        self.assertEqual(sent_message.data['page'], [image_url])
```
[the remark this answers is not in the document]
Comment (forslund) [reply test/unittests/enclosure/test_gui.py]: Good suggestion. Should be updated now.

**Comment by forslund** [comment]
Thanks for the quick review. I've pushed updates and the issues should be resolved.

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2697))

**Review by krisgesling — APPROVED** [review]

**Pull request merged**